code/modules/food_and_drinks/kitchen_machinery/kitchen_machine.dm

```diff
@@ -439,3 +446,20 @@
 		if("dispose")
 			dispose()
 	return
+
+/obj/machinery/kitchen_machine/proc/build_ingredient_description()
+	if(contents)
+		var/list/contents_strings
+		var/list/item_counts
+		for(var/datum/I in contents)
+			item_counts[I.type] += 1
+		for(var/T in item_counts)
+			contents_strings += ""
+
+/obj/machinery/kitchen_machine/examine(mob/user, infix = "", suffix = "")
+	. = build_base_description(infix, suffix)
+	var/dat = format_content_descs()
+	if(dat)
+		. += "It contains: <BR>[dat]"
+
+	SEND_SIGNAL(src, COMSIG_PARENT_EXAMINE, user, .)
```

Comment:
This is not the best way to go about this. You've skipped the parent call which means kitchen machines miss out on the things in `/obj/machinery/examine` and you're adding another instance of `COMSIG_PARENT_EXAMINE` being sent which should be avoided.

Since you've done this to avoid calling `build_reagent_description` (I think), instead what you should do is this:
```dm
/obj/machinery/kitchen_machine/examine(mob/user, infix = "", suffix = "")
	. = ..()
	var/dat = format_content_descs()
	if(dat)
		. += "It contains: <BR>[dat]"

/obj/machinery/kitchen_machine/build_reagent_description(mob/user)
	return
```

Reply:
Yeah, what I was trying to do here was have all the tool's description available before the signal was sent, which is why the parent call wasn't made (therefore skipping /atom/proc/examine's signal), but it slipped my mind that none of the intermediate object types would have their examines called either. I think I cleaned this up properly.